### ISSUE TITLE
SignalR error: Error during negotiation request. Handshake failed

### ISSUE THREAD

**Issue by emirsnck** [olofd/react-native-signalr#23]
Hi, I'm getting an error
My backend code
```
    public class FirstHub : Hub, IDisposable
    {

        public string HelloServer(string message)
        {
            Clients.Caller.helloApp("You said:", message, "Server says hello!", (DateTime.Now).ToString(CultureInfo.InvariantCulture));
            return "I responeded by calling helloApp";
        }

        public override Task OnConnected()
        {
            return base.OnConnected();
        }

        public override Task OnReconnected()
        {
            return base.OnReconnected();
        }

        public override Task OnDisconnected(bool stopCalled)
        {
            return base.OnDisconnected(stopCalled);
        }
}
```
startup.cs
```
[assembly: OwinStartupAttribute(typeof(Project.Startup))]
namespace Project
{
    public class Startup
    {
        public void Configuration(IAppBuilder app)
        {
            app.MapSignalR();
        }
    }
}

```

react native code:
```
componentDidMount() {
    //This is the server under /example/server published on azure.
    const connection = signalr.hubConnection('https://10.26.xx.xx:xxxx');
    connection.logging = true;

    const proxy = connection.createHubProxy('firstHub');
    //receives broadcast messages from a hub function, called "helloApp"
   proxy.on('helloApp', (argOne, argTwo, argThree, argFour) => {
      console.log('message-from-server', argOne, argTwo, argThree, argFour);
      //Here I could response by calling something else on the server...
    });

    // atempt connection, and handle errors
    connection.start().done(() => {
      console.log('Now connected, connection ID=' + connection.id);

      proxy.invoke('helloServer', 'Hello Server, how are you?')
        .done((directResponse) => {
          console.log('direct-response-from-server', directResponse);
        }).fail(() => {
          console.warn('Something went wrong when calling server, it might not be up and running?')
        });

    }).fail(() => {
      console.log('Failed');
    });

    //connection-handling
    connection.connectionSlow(() => {
      console.log('We are currently experiencing difficulties with the connection.')
    });

    connection.error((error) => {
      const errorMessage = error.message;
      let detailedError = '';
      if (error.source && error.source._response) {
        detailedError = error.source._response;
      }
      if (detailedError === 'An SSL error has occurred and a secure connection to the server cannot be made.') {
        console.log('When using react-native-signalr on ios with http remember to enable http in App Transport Security https://github.com/olofd/react-native-signalr/issues/14')
      }
      console.debug('SignalR error: ' + errorMessage, detailedError)
    });
  }
```

react native chrome debuger console error log ;

![capture](https://cloud.githubusercontent.com/assets/9586282/24607505/cfb73694-187a-11e7-9d75-a2810666d376.PNG)

**Comment by jp928** [comment]
On server side, try to enable CORS.

**Comment by emirsnck** [comment]
i solved this problem ,
i changed https to http and 10.26.12 this ip permission reach to localhost and its work :)

**Comment by olofd** [comment]
FYI. New version just published 1.0.4.
